packages/mdc-textfield/helper-text/README.md

```diff
@@ -0,0 +1,133 @@
+<!--docs:
+title: "Text Field Helper Text"
+layout: detail
+section: components
+excerpt: "The helper text provides supplemental information and/or validation messages to users"
+iconId: text_field
+path: /catalog/input-controls/text-fields/helper-text/
+-->
+
+# Text Field Helper Text
+
+The helper text provides supplemental information and/or validation messages to users. It appears on input field focus and disappears on input field blur by default, or it can be persistent.
+
+## Design & API Documentation
+
+<ul class="icon-list">
+  <li class="icon-list-item icon-list-item--spec">
+    <a href="https://material.io/guidelines/components/text-fields.html#text-fields-layout">Material Design guidelines: Text Fields Layout</a>
+  </li>
+</ul>
+
+
+## Usage
+
+### Using helper text
+
+MDC Text Fields can include helper text that is useful for providing supplemental
+information to users, as well for validation messages (covered below).
+
+```html
+<div class="mdc-text-field">
+  <input type="text" id="username" class="mdc-text-field__input" aria-controls="username-helper-text">
+  <label for="username" class="mdc-text-field__label">Username</label>
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+<p id="username-helper-text" class="mdc-text-field-helper-text" aria-hidden="true">
+  This will be displayed on your public profile
+</p>
+```
+
+Helper text appears on input field focus and disappears on input field blur by default when using
+the Text Field JS component.
+
+#### Persistent helper text
+
+If you'd like the helper text to always be visible, add the `mdc-text-field-helper-text--persistent` modifier class to the element.
+
+```html
+<div class="mdc-text-field">
+  <input type="email" id="email" class="mdc-text-field__input">
+  <label for="email" class="mdc-text-field__label">Email address</label>
+  <div class="mdc-text-field__bottom-line"></div>
+</div>
+<p class="mdc-text-field-helper-text mdc-text-field-helper-text--persistent">
+  We will <em>never</em> share your email address with third parties
+</p>
+```
+
+#### Helper text and accessibility
+
+Note that in every example where the helper text is dependent on the state of the input element, we
+assign an id to the `mdc-text-field-helper-text` element and set that id to the value of the
```

Comment:
Can you add a code snippet example demonstrating the id concept?

Reply:
This code snippet would be exactly the same as the the first example snippet on the page. Should I remove the `id`/`aria-controls` attributes from the first snippet since they are technically optional, or let it be duplicate?

Reply:
I think you should remove it from the first snippet...but open to suggestions

Reply:
sounds good to me